### PR TITLE
Local UI

### PR DESCRIPTION
**Pull request by akhilgupta1093**
Connected to this issue: https://github.com/Ribbit-Network/ribbit-network-frog-software/issues/10

Create a local UI to monitor sensor status and present a bit more information about the sensor. UI looks like this--

Frog Sensor Status

Open to feedback!

### CODE REVIEW

**Comment by keenanjohnson** [comment]
Wow this is excellent! I'm traveling for the next few days, but will take a look soon! Thank you for the hardwork!

What features have you implemented here?

**Comment by akhilgupta1093** [comment]
It shows the status of each of the sensors and the corresponding information if the sensor is online. At the bottom, it shows a little extra information like the wifi network, golioth host, and golioth user. 

Seems like I developed this with a slightly stale version of the code, so I need to do some updates to make it work again! 

One thing I'm figuring out is the mdns domain. Right now, the domain is espressif, so "espressif.local" routes to the sensor's webpage, but I'm figuring out how to change that.

**Comment by akhilgupta1093** [comment]
Cool, updated! Should work now with the newest data model. I might delay getting frog.local to work to a different MR.

**Comment by keenanjohnson** [comment]
Great start @akhilgupta1093 ! I've confirmed this runs nicely on my Frog here: 
image

I noticed a few small things I will comment on in the code to tweak before merging.

**Review by keenanjohnson — CHANGES_REQUESTED** [review]
I left a few comments of things to fix up:

- [ ] Removing the vscode files
- [ ] Updating the page title
- [ ] I noticed that the micropython submodule was updated. That doesn't seem intentional and should be removed?

Additionally, the sensor update rate seems a bit slow to me. Perhaps we should increase it? What do you think?

File: modules/ribbit/http.py
Comment (damz) [inline modules/ribbit/http.py:53]: I think polling on the client side makes more sense here than websockets.

But, if we want websockets:

* This loop probably wants a `await asyncio.sleep_ms(xxxx)` somewhere?
* It would be great to just store the last measured value somewhere instead of polling the sensors directly

File: modules/ribbit/http.py
Comment (damz) [inline modules/ribbit/http.py:55]: This probably should stay `/api/config`.

File: modules/ribbit/http.py
Comment (damz) [inline modules/ribbit/http.py:76]: Why are you removing this?

File: Makefile
Comment (damz) [inline Makefile:33]: Please revert this.

**Review by damz — COMMENTED** [review]
Adding a few comments of my own.

I would recommend we do not commit the generated `modules/ribbit/_static.py` file, and instead run `modules/ribbit/generate_static.py` (which should be moved elsewhere because it is not something we need on the frog itself) at build time in the Github action.

File: Makefile
Comment (keenanjohnson) [reply Makefile:33]: I wonder what the best way to handle this is. I guess conditionals based on the operating system? 

I have to change it on my machine as well (also a macbook) to get this to flash. Probably not worth resolving in this review, but it is perhaps worth opening a separate issue to make the Makefile more robust to this.

**Comment by akhilgupta1093** [comment]
Thanks for the feedback! I'll take a look at it within a few days. @keenanjohnson I made it so the "last updated" time only updates when the values change, but I understand now that that's not super intuitive. Let me change it so the time updates every time we get new data.

File: modules/ribbit/http.py
Comment (akhilgupta1093) [reply modules/ribbit/http.py:76]: Hmm, I can't seem to find where this route is being used. What is this used for?

File: modules/ribbit/http.py
Comment (akhilgupta1093) [reply modules/ribbit/http.py:55]: If this route is /api/config and the @app.patch('api/config') method remains below, will that conflict?

File: modules/ribbit/http.py
Comment (akhilgupta1093) [reply modules/ribbit/http.py:53]: Totally, I think I overcomplicated things by polling the device directly as well. Now planning to just get the desired values directly from the sensor (they're already stored, for dps310 it's self.temperature and self.pressure, etc.).

Comment (akhilgupta1093) [reply modules/ribbit/http.py:53]: What do you mean by "polling on the client side"?

**Comment by akhilgupta1093** [comment]
@keenanjohnson @damz Made some changes and added some replies, please take a look when you get the chance, thanks!

File: modules/ribbit/http.py
Comment (damz) [reply modules/ribbit/http.py]: This is the API to mutate the config.

File: modules/ribbit/http.py
Comment (damz) [reply modules/ribbit/http.py:80]: They will not conflict: one is `GET`, the other is `PATCH`.

File: modules/ribbit/http.py
Comment (damz) [reply modules/ribbit/http.py:78]: I mean just doing an API request regularly (via `setInterval()`).

Comment (damz) [reply modules/ribbit/http.py:78]: Not only it is simpler, it also avoid having to store state on the device side, which is quite limited in memory.

**Comment by keenanjohnson** [comment]
Looks like all of my comments were addressed except for the submodule change. If you merge my commit from #26 into this branch, that should resolve that though.

Great work team!

File: modules/ribbit/http.py
Comment (akhilgupta1093) [reply modules/ribbit/http.py:75]: Because I feel like I have limited bandwidth, perhaps we can use websockets for now and iterate on this point if we so desire

**Comment by akhilgupta1093** [comment]
@keenanjohnson @damz Updated based on your comments! One thing-- if possible, we can keep the websocket code for now and iterate to an api request with `setInterval()` like Damien suggested in the future.

**Review by keenanjohnson — APPROVED** [review]
All the changes I had requested are made, so I think this looks great to merge!

**Comment by keenanjohnson** [comment]
I had selected web sockets in my original testing as a somewhat arbitrary choice originally in my example, but definitely happy to switch it later if that seems to be a better technical choice for this environment.

However, I'm in favor of merging this now which would allow us to have a first useful base to start iterating on.

**Comment by keenanjohnson** [comment]
Pending @damz opinion, I can do the merge whenever you are ready @akhilgupta1093 .

**Comment by akhilgupta1093** [comment]
I'm good to merge once Damien gives the go ahead as well

**Pull request merged**
